src/bootstrap/bootstrap.py

```diff
@@ -760,7 +763,13 @@ def main():
     help_triggered = (
         '-h' in sys.argv) or ('--help' in sys.argv) or (len(sys.argv) == 1)
     try:
-        bootstrap()
+        # If the user is asking for help, let them know that the whole download-and-build
+        # process has to happen before anything is printed out.
+        if help_triggered:
+            print("NOTE: Downloading and compiling bootstrap requirements before processing")
+            print("      --help command. See src/bootstrap/README.md for help with common")
+            print("      commands.")
```

Comment:
We should get this message only if we actually going to download something or do other long operation.

Also there's already `print_what_bootstrap_means` that is supposed to print this exact information, but maybe it doesn't fire for some reason?

Reply:
I decided to print the message every time, because we _always_ rebuild the build tool. So it will _never_ be instant. Also, doing anything more complicated than printing this at the beginning means that we have to go through the whole process to gather information, setup environment variables, and check if anything needs to be done. It would require a lot of logic spread through a lot of the build script, and that complexity simply isn't worth it.

I have never noticed `print_what_bootstrap_means`. Looking back at it, it only triggers if a download happens, and is _immediately_ followed by a lot of download messages and loading bars. Also, prior to this fix, the very first thing the build script did was `update_submodules`, so `print_what_bootstrap_means` would be printed after the most time consuming part of the build script.

The message also doesn't reference --help at all; it states why stuff needs to download before running build commands, but the user will not expect --help to be part of that. My message makes that perfectly clear.

Basically, we either have to print this whenever --help happens, regardless of what we actually need to do; OR we have to wait until the build script sets up enough to _know_ if it needs to download, at which point seconds have already passed and the tool will feel sluggish (asking the user to wait after the user has already waited). And if we want to do the second option, because of how the script is organized, we would either have to do a lot of refactoring, or litter the code with checks and print statements.

Reply:
Ok, sounds good.
Could you remove `print_what_bootstrap_means` (and also `self.printed`) then?